go/cmd/dolt/dolt.go

```diff
@@ -609,6 +609,12 @@ func buildLateBinder(ctx context.Context, cwdFS filesys.Filesys, mrEnv *env.Mult
 	var targetEnv *env.DoltEnv = nil
 
 	useDb, hasUseDb := apr.GetValue(commands.UseDbFlag)
+	useBranch, hasBranch := apr.GetValue(cli.BranchParam)
+
+	if hasUseDb && hasBranch{
+		dbName, _ := dsess.SplitRevisionDbName(useDb)
```

Comment:
When a user provide a branch in the DB and with the flag, they will get confusing results. You should error out when there is a branch in the db name that will be ignored.

Reply:
made the changes